libifupdown/execute.c

```diff
@@ -30,6 +30,31 @@
 
 #define SHELL	"/bin/sh"
 
+/* TODO: Add support for Linux process descriptors once it is okay to require
+ * Linux 5.3 or newer.
+ */
+static inline bool
+lif_process_monitor(const char *cmdbuf, pid_t child, int timeout_sec)
+{
+	int status;
+	int ticks = 0;
+
+	while (ticks < timeout_sec)
+	{
+		if (waitpid(child, &status, 0) == child)
```

Comment:
Won't this block/hang until the child has stopped so the sleep() + counter aren't used at all? I would have expected `WNOHANG` as option here, am I missing something?

Reply:
Forgot to actually add the `WNOHANG` :P